check.go

```diff
@@ -182,6 +188,20 @@ type CheckGitBranchProtectionUpdateInput struct {
 	CheckUpdateInput
 }
 
+type CheckHasDocumentationCreateInput struct {
+	CheckCreateInput
+
+	DocumentType    HasDocumentationTypeEnum    `graphql:"documentType"`
```

Comment:
because these are "Input" they will be run through the JSON marshaler to go into the `variables` and that uses the `json` struct tags not the `graphql` struct tags so it needs to be

```
DocumentType    HasDocumentationTypeEnum    `json"documentType"`
```

Only the "return" types IE `HasDocumentationCheckFragment` needs the `graphql` struct tags because the graphql framework is marshaling the response from our API back into that struct and those tags tell it how to map the json back to the fields.

This is a very golang centric thing about data marshaling and structs - they are basically notes for the different marshalers to do their work and since we have to PascalCase the field names on the golang side so they are accessible by an enduser of this library we have to apply the struct tags to everything. :(

Reply:
Ah ok thanks again for this context! I was having a hard time figuring out why this was happening.